test/expiring-todo-comments.js

```
@@ -0,0 +1,173 @@
+			output: '// TODO [-popura]: when you uninstall `popura`'
+		},
+		{
+			code: '// TODO [read-pkg@>1]: when `read-pkg` version is > 1',
+			errors: [versionMatchesError('read-pkg > 1')],
+			output: '// TODO [read-pkg@>1]: when `read-pkg` version is > 1'
+		},
+		{
+			code: '// TODO [engines node>=8]: when support is for node > 8',
+			errors: [engineMatchesError('node >= 8')],
+			output: '// TODO [engines node>=8]: when support is for node > 8'
+		},
+		{
+			code: '// TODO [read-pkg@>=5.1.1]: when `read-pkg` version is >= 5.1.1',
+			errors: [versionMatchesError('read-pkg >= 5.1.1')],
+			output: '// TODO [read-pkg@>=5.1.1]: when `read-pkg` version is >= 5.1.1'
+		},
+		{
+			code: '// TODO [> 1]: Default rule if space',
+			errors: [noWarningCommentError('todo')],
+			output: '// TODO [> 1]: Default rule if space'
+		},
+		{
+			code: '// TODO [semver>1]: Default rule if no @.',
+			errors: [noWarningCommentError('todo')],
+			output: '// TODO [semver>1]: Default rule if no @.'
+		},
+		{
+			code: '// TODO [semver@> 1]: Default rule if space.',
+			errors: [noWarningCommentError('todo')],
+			output: '// TODO [semver@> 1]: Default rule if space.'
+		},
+		{
+			code: '// TODO [semver @>1]: Default rule if space.',
+			errors: [noWarningCommentError('todo')],
+			output: '// TODO [semver @>1]: Default rule if space.'
+		},
+		{
+			code: '// TODO [semver@>= 1]: Default rule if space.',
+			errors: [noWarningCommentError('todo')],
+			output: '// TODO [semver@>= 1]: Default rule if space.'
+		},
+		{
+			code: '// TODO [semver @>=1]: Default rule if space.',
+			errors: [noWarningCommentError('todo')],
+			output: '// TODO [semver @>=1]: Default rule if space.'
+		},
+		{
+			code: '// TODO [engines node >=1]: Default rule if space.',
+			errors: [noWarningCommentError('todo')],
+			output: '// TODO [engines node >=1]: Default rule if space.'
+		},
+		{
+			code: '// TODO [engines node>= 1]: Default rule if space.',
```

This should fail loud and not silently, and inform the user what they did wrong.

---

Done